code/modules/research/xenobiology/crossbreeding/_structures.dm

```diff
@@ -0,0 +1,667 @@
+	. = ..()
+	if(!istype(affected_mob,/mob/living/carbon))
+		return
+	var/mob/living/carbon/carbon_mob = affected_mob
+	var/rand_dam_type = rand(0,10)
+
+	switch(rand_dam_type)
+		if(0)
+			carbon_mob.adjustBruteLoss(-heal_amt)
+		if(1)
+			carbon_mob.adjustFireLoss(-heal_amt)
+		if(2)
+			carbon_mob.adjustOxyLoss(-heal_amt)
+		if(3)
+			carbon_mob.adjustToxLoss(-heal_amt)
+		if(4)
+			carbon_mob.adjustCloneLoss(-heal_amt)
+		if(5)
+			carbon_mob.adjustStaminaLoss(-heal_amt)
+		if(6 to 10)
+			carbon_mob.adjustOrganLoss(pick(ORGAN_SLOT_BRAIN,ORGAN_SLOT_HEART,ORGAN_SLOT_LIVER,ORGAN_SLOT_LUNGS),-heal_amt)
+
+/obj/structure/slime_crystal/blue
+	colour = "blue"
+	range_type = "view"
+
+/obj/structure/slime_crystal/blue/process()
+	for(var/turf/open/T in view(2,src))
+		if(isspaceturf(T))
+			continue
+		var/datum/gas_mixture/gas = T.return_air()
+		gas.parse_gas_string(OPENTURF_DEFAULT_ATMOS)
+		T.air_update_turf()
+
+/obj/structure/slime_crystal/metal
+	colour = "metal"
+
+	var/heal_amt = 1
+
+/obj/structure/slime_crystal/metal/on_mob_effect(mob/living/affected_mob)
+	. = ..()
+	if(!iscyborg(affected_mob))
+		return
+	var/mob/living/silicon/borgo = affected_mob
+	borgo.adjustBruteLoss(-heal_amt)
+
+/obj/structure/slime_crystal/yellow
+	colour = "yellow"
+	light_color = LIGHT_COLOR_YELLOW //a good, sickly atmosphere
+	light_power = 0.75
+
+/obj/structure/slime_crystal/yellow/Initialize()
+	. = ..()
+	set_light(3)
+
+/obj/structure/slime_crystal/yellow/attacked_by(obj/item/I, mob/living/user)
+	if(istype(I,/obj/item/stock_parts/cell))
+		var/obj/item/stock_parts/cell/cell = I
+		//Punishment for greed
+		if(cell.charge == cell.maxcharge)
+			cell.explode()
+			return
+		cell.give(cell.maxcharge)
+		return
+	return ..()
+/obj/structure/slime_crystal/darkpurple
+	colour = "dark purple"
+
+/obj/structure/slime_crystal/darkpurple/process()
+	var/turf/T = get_turf(src)
+	if(!istype(T,/turf/open))
+		return
+	var/turf/open/open_turf = T
+	var/datum/gas_mixture/air = open_turf.return_air()
+
+	if(air.get_moles(/datum/gas/plasma) > 15)
+		air.adjust_moles(/datum/gas/plasma,-15)
+		new /obj/item/stack/sheet/mineral/plasma(open_turf)
+
+/obj/structure/slime_crystal/darkpurple/Destroy()
+	atmos_spawn_air("plasma=[20];TEMP=[500]")
+	return ..()
+
+/obj/structure/slime_crystal/darkblue
+	colour = "dark blue"
+
+/obj/structure/slime_crystal/darkblue/process()
+	var/list/listie = range(5,src)
+	for(var/turf/open/T in listie)
+		if(prob(75))
+			continue
+		var/turf/open/open_turf = T
+		open_turf.MakeDry(TURF_WET_LUBE)
+
+	for(var/obj/item/trash/trashie in listie)
+		if(prob(75))
+			continue
+		qdel(trashie)
+
+/obj/structure/slime_crystal/silver
+	colour = "silver"
+
+/obj/structure/slime_crystal/silver/process()
+	for(var/obj/machinery/hydroponics/hydr in range(5,src))
+		hydr.weedlevel = 0
+		hydr.pestlevel = 0
+		hydr.age++
+
+/obj/structure/slime_crystal/bluespace
+	colour = "bluespace"
+	density = FALSE
+	///Is it in use?
+	var/in_use = FALSE
```

Comment:
Since these don't do anything on process() you should have something that allows you to make it not call process in the first place and waste processing time doing nothing